src/backend/src/services/teams.services.ts

```diff
@@ -138,15 +138,15 @@ export default class TeamsService {
     if (!isHead(newHead.role)) throw new AccessDeniedException('The team head must be at least an head');
 
     const newHeadTeam = await prisma.team.findFirst({
```

Comment:
this is kinda confusing im realizing, can you put a comment before this saying, checking to see if any teams have the new head as their current head